src/Core/Form/IdentifiableObject/CommandBuilder/Product/SpecificPricePriorityCommandsBuilder.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\RemoveSpecificPricePriorityForProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\SetSpecificPricePriorityForProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+class SpecificPricePriorityCommandsBuilder implements ProductCommandsBuilderInterface
+{
+    public function buildCommands(ProductId $productId, array $formData): array
+    {
+        if (!isset($formData['pricing']['priority_management'])) {
+            return [];
+        }
+
+        $prioritiesData = $formData['pricing']['priority_management'];
+
+        if (empty($prioritiesData['use_custom_priority'])) {
```

Comment:
you consider that when $formData['pricing']['priority_management'] is defined alaways $formData['pricing']['priority_management']['use_custom_priority'] is also .... dangerous ?

Reply:
you check if empty but data is init with a FALSE in [ProductFormDataProvider.php]:214

Reply:
not sure if i get what are you asking, but this should be alright.
use_custom_priority is a checkbox in UI, which once checked allows defining custom priorities for product.
In this case, if its unchecked (either false, or for some reason not set at all) I assume we use default priorites (which is achieved by simply deleting all the custom priorities for this product).